File: src/main/java/com/indiepost/model/Bookmark.java
package com.indiepost.model;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * Created by jake on 7/25/16.
 */
@Entity
@Table(name = "Bookmarks")
public class Bookmark implements Serializable {

    private static final long serialVersionUID = 5793341286387473902L;

    @Id
    @ManyToOne
    @JoinColumn(name = "userId")
    private User user;

    @Id
    @ManyToOne
    @JoinColumn(name = "postId")
    private Post post;

    @Column(nullable = false)
    private LocalDateTime createdAt;

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Post getPost() {
        return post;
    }

    public void setPost(Post post) {
        this.post = post;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: src/main/kotlin/com/indiepost/model/Bookmark.kt
package com.indiepost.model

import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "Bookmarks")
data class Bookmark(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        var id: Long? = null,

        @Column(nullable = false)
        var created: LocalDateTime? = null,
        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "postId", nullable = false)
        var post: Post? = null,

        @Column(name = "postId", insertable = false, updatable = false)
        var postId: Long? = null,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "userId", nullable = false)
        var user: User? = null,

        @Column(name = "userId", insertable = false, updatable = false)
        var userId: Long? = null
)
